test/system/272-system-connection.bats

```diff
@@ -180,4 +180,72 @@ $c2[ ]\+tcp://localhost:54321[ ]\+true" \
     run_podman system connection rm mysshconn
 }
 
+@test "podman-remote: non-default connection" {
+    # priority:
+    #   1. cli flags (--connection ,--url ,--context ,--host)
+    #   2. Env variables (CONTAINER_HOST and CONTAINER_CONNECTION)
+    #   3. ActiveService from containers.conf
+    #   4. RemoteURI
+
+    # setup
+    run_podman system connection add test1 unix:/run/user/test1/podman/podman.sock   # default=true
+    run_podman system connection add test2 unix:/run/user/test2/podman/podman.sock
+    run_podman system connection add test3 unix:/run/user/test3/podman/podman.sock
+
+    # Test priority of Env variables wrt cli flags
+    # test env variable CONTAINER_CONNECTION wrt --connection cli flag
+    CONTAINER_CONNECTION=test2 run_podman 125 --connection=test3 ps
+    assert "$output" =~ ".*/run/user/test3/podman/podman.sock.*" "Run Podman with connection flag and env variable CONTAINER_CONNECTION"
```

Comment:
Hint: you don't need `.*` when using `=~`. (Clarification: not at beginning or end. You still need it in the bottommost one. Like, `=~ ".*foo.*"` should just be `=~ "foo"`, but `=~ "foo.*bar"` stays as-is)